http://sourceware.org/ml/gdb-cvs/2013-06/msg00044.html

### src/gdb/testsuite/ChangeLog	2013/06/06 19:00:13	1.3688
### src/gdb/testsuite/ChangeLog	2013/06/06 19:02:26	1.3689
## -1,3 +1,7 @@
+2013-06-06  Doug Evans  <dje@google.com>
+
+	* gdb.cp/derivation.exp: Make tests have unique names.
+
 2013-06-06  Tom Tromey  <tromey@redhat.com>
 
 	* gdb.base/dump.exp (capture_value): Don't put expression into
--- src/gdb/testsuite/gdb.cp/derivation.exp	2013/06/05 22:28:51	1.25
+++ src/gdb/testsuite/gdb.cp/derivation.exp	2013/06/06 19:02:27	1.26
@@ -40,15 +40,17 @@
 }
 
 # Check inheritance of typedefs.
-foreach klass {"A" "D" "E" "F" "A2" "D2"} {
-    gdb_test "ptype ${klass}::value_type" "type = int"
-    gdb_test "whatis ${klass}::value_type" "type = int"
-    gdb_test "p (${klass}::value_type) 0" " = 0"
-}
-foreach klass {"Z" "ZZ"} {
-    gdb_test "ptype ${klass}::value_type" "type = float"
-    gdb_test "whatis ${klass}::value_type" "type = float"
-    gdb_test "p (${klass}::value_type) 0" " = 0"
+with_test_prefix "before run" {
+    foreach klass {"A" "D" "E" "F" "A2" "D2"} {
+	gdb_test "ptype ${klass}::value_type" "type = int"
+	gdb_test "whatis ${klass}::value_type" "type = int"
+	gdb_test "p (${klass}::value_type) 0" " = 0"
+    }
+    foreach klass {"Z" "ZZ"} {
+	gdb_test "ptype ${klass}::value_type" "type = float"
+	gdb_test "whatis ${klass}::value_type" "type = float"
+	gdb_test "p (${klass}::value_type) 0" " = 0"
+    }
 }
 
 # Set it up at a breakpoint so we can play with the variable values.
@@ -59,10 +61,12 @@
 }
 
 # Check inheritance of typedefs again, but this time with an active block.
-foreach klass {"A" "D" "A2" "D2"} {
-    gdb_test "ptype ${klass}::value_type" "type = int"
-    gdb_test "whatis ${klass}::value_type" "type = int"
-    gdb_test "p (${klass}::value_type) 0" " = 0"
+with_test_prefix "at marker1" {
+    foreach klass {"A" "D" "A2" "D2"} {
+	gdb_test "ptype ${klass}::value_type" "type = int"
+	gdb_test "whatis ${klass}::value_type" "type = int"
+	gdb_test "p (${klass}::value_type) 0" " = 0"
+    }
 }
 
 gdb_test "up" ".*main.*" "up from marker1"
